impeller/compiler/shader_lib/impeller/types.glsl

```diff
@@ -9,6 +9,12 @@
 #extension GL_AMD_gpu_shader_half_float_fetch : enable
 #extension GL_EXT_shader_explicit_arithmetic_types_float16 : enable
 
+#ifdef IMPELLER_TARGET_OPENGLES
+#define IMPELLER_FLAT
+#else
+#define IMPELLER_FLAT flat
```

Comment:
We should call thin `IMPELLER_MAYBE_FLAT`. Right now, we only use flat as an optimization to reduce load on the varying unit. But someone might actually start using it in the future to disregard non-provoking vertices and will be confused as to why it doesn't work on OpenGL ES.

Reply:
Done